os_net_config/objects.py

```diff
@@ -404,10 +406,17 @@ def __init__(self, name, use_dhcp=False, use_dhcpv6=False, addresses=None,
                 self.renamed = True
             else:
                 self.name = '%s%s' % (mapped_nic_names[base_name], vlan_suffix)
+            if mtu is not None:
+                interface = mapped_nic_names[base_name]
+                max = utils.get_interface_maxmtu(interface)
+                if (max != -1):
+                    ismaxmtu = (mtu > max)
         else:
             self.name = name
+            max = utils.get_interface_maxmtu(name)
+            if (max != -1):
```

Comment:
This block could be moved outside the else {}
if max_mtu != -1 and mtu > max_mtu:
            msg = f"{name}: desired MTU {mtu} is greater than supported {max_mtu}"
            raise InvalidConfigException(msg)